common/src/main/java/com/wynntils/features/user/overlays/InfoBoxesFeature.java

```diff
@@ -0,0 +1,218 @@
+                            .withTextShadow(textShadow));
+        }
+
+        private String parseColorCodes(String toProcess) {
+            // For every & symbol, check if the next symbol is a color code and if so, replace it with §
+            // But don't do it if a \ precedes the &
+            String validColors = "0123456789abcdefklmnor";
+            StringBuilder sb = new StringBuilder(toProcess);
+            for (int i = 0; i < sb.length(); i++) {
+                if (sb.charAt(i) == '&') { // char == &
+                    if (i + 1 < sb.length()
+                            && validColors.contains(String.valueOf(sb.charAt(i + 1)))) { // char after is valid color
+                        if (i - 1 < 0 || sb.charAt(i - 1) != '\\') { // & is first char || char before is not \
+                            sb.setCharAt(i, '§');
+                        } else if (sb.charAt(i - 1) == '\\') { // & is preceded by \, just remove the \
+                            sb.deleteCharAt(i - 1);
+                        }
+                    }
+                }
+            }
+            return sb.toString();
+        }
+
+        private String doEscapeFormat(String escaped) {
+            return switch (escaped) {
+                case "\\" -> "\\\\";
+                case "n" -> "\n";
+                case "%" -> "%";
+                case "§" -> "&";
+                case "E" -> EmeraldSymbols.E_STRING;
+                case "B" -> EmeraldSymbols.B_STRING;
+                case "L" -> EmeraldSymbols.L_STRING;
+                case "M" -> "✺";
+                case "H" -> "❤";
+                default -> null;
+            };
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {
+            toRender = recalculateFunctions(content);
+        }
+
+        @Override
+        public void render(PoseStack poseStack, float partialTicks, Window window) {
+            if (!WynnUtils.onWorld()) return;
+            FontRenderer.getInstance()
+                    .renderTextWithAlignment(
+                            poseStack,
+                            this.getRenderX(),
+                            this.getRenderY(),
+                            toRender,
+                            this.getWidth(),
+                            this.getHeight(),
+                            this.getRenderHorizontalAlignment(),
+                            this.getRenderVerticalAlignment());
+        }
+
+        @Override
+        public void renderPreview(PoseStack poseStack, float partialTicks, Window window) {
+            if (!WynnUtils.onWorld()) return;
+
+            // FIXME: We do re-calculate this on render, but this is preview only, and fixing this would need a lot of
+            // architectural changes at the moment
+            TextRenderTask toRenderPreview = recalculateFunctions("&cX: %x%, &9Y: %y%, &aZ: %z%");
```

Comment:
I don't get this. How does the normal toRender get updated? Only on config changes? Even if it does not need recalculating on every rendered frame, it do need some periodic update.. Or am I missing something?

Reply:
Wait..., I need to do this in toRender too otherwise we are getting old data. Well..

Reply:
Most functions are relatively cheap, but some iterate through all inventory parsing lore, and that's no good to make for each frame we're drawing.